contracts/core/Dispatcher.sol

```diff
@@ -438,6 +417,25 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         emit Timeout(address(receiver), packet.src.channelId, packet.sequence);
     }
 
+    function addNewLightClient(string calldata connectionId, uint256 clientId, LightClient newLightClient)
+        external
+        onlyOwner
+    {
+        if (address(newLightClient) == address(0)) {
+            revert IBCErrors.invalidInput();
+        }
+        if (clientId == 0 || bytes(connectionId).length == 0) {
+            revert IBCErrors.invalidInput();
+        }
+
+        if (connectionIdToClientId[connectionId] != 0 || clientIdToLightClient[clientId] != LightClient(address(0))) {
+            revert IBCErrors.invalidInput();
+        }
+
+        connectionIdToClientId[connectionId] = clientId;
+        clientIdToLightClient[clientId] = newLightClient;
```

Comment:
As per Alexander's comments on https://github.com/open-ibc/vibc-core-smart-contracts/pull/48 it seems like having a single mapping of complex types is more efficient than multiple maps of simple types. An option will be to merge these two maps into one

Reply:
yes! Planning to implement this in the Proxy design PR

Reply:
That is only true when the struct can be packed in fewer storage slots. For example, in the other case you were using 3x1 byte (3 bools) which obviously fit in 32 bytes.
I am not sure these can be compressed further.